src/TextForEvent.tsx

```diff
@@ -953,20 +958,29 @@ export function hasText(ev: MatrixEvent, client: MatrixClient, showHiddenEvents?
  * @param allowJSX Whether to output rich JSX content
  * @param showHiddenEvents An optional cached setting value for showHiddenEventsInTimeline
  *     to avoid hitting the settings store
+ * @param enclosedSender An optional parameter to check the sender name should be enclosed within
+ *      < > brackets or not
  */
 export function textForEvent(ev: MatrixEvent, client: MatrixClient): string;
+export function textForEvent(
+    ev: MatrixEvent,
+    client: MatrixClient,
+    enclosedSender: true,
+): string;
```

Comment:
the problem here is that the new overload signature doesn't match the real definition of the function: the third parameter is always `allowJSX`.

https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads has more information on overload signatures, and explains it better than I can.